src/roiextractors/extractors/inscopixextractors/inscopixsegmentationextractor.py

```diff
@@ -38,25 +44,92 @@ def __init__(self, file_path: PathType):
 
         SegmentationExtractor.__init__(self)
         self.file_path = file_path
-        self.cell_set = isx.CellSet.read(file_path)
+        file_path_str = str(file_path)
+
+        self.cell_set = isx.CellSet.read(file_path_str, read_only=True)
```

Comment:
I think this should be a private attribute by default. We can think later on making it public if it makes sense. In general, if you don't want the burden of not being able to change things, they should start as a private.

What do you think?